fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1589,6 +1605,168 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
         return rootParameter;
     }
 
+    /**
+     * Transforms the passed string representing reverse chain search criteria, into
+     * an actual chain of QueryParameter objects. This method consumes strings of this form:
+     * <pre>
+     *      +-------------------------------------------------------------------+
+     *      |                                                                   |
+     *      V                                                                   |
+     * >>---+--- "_has:{referenced-by-resource-type}:{reference-parameter}:" ---+--- "{search-parameter}" ---><
+     * </pre>
+     * See the FHIR specification for details:
+     * <a href="https://www.hl7.org/fhir/search.html#has</a>
+     *
+     * @param resourceType
+     *          Search type.
+     * @param reverseChainParameterString
+     *          Reverse chain search parameter string.
+     * @param valuesString
+     *          String containing the final search value.
+     * @return QueryParameter
+     *          The root of a parameter chain for the reverse chain criteria.
+     */
+    private static QueryParameter parseReverseChainedParameter(Class<?> resourceType, String reverseChainParameterString, String valuesString) throws Exception {
+
+        QueryParameter rootParameter = null;
+
+        try {
+            // Strip leading '_has:' and then split by ':_has:'
+            List<String> components = Arrays.asList(reverseChainParameterString
+                .replaceFirst(SearchConstants.HAS + SearchConstants.COLON_DELIMITER_STR, "")
```

Comment:
I think substring might be preferred here instead of replaceFirst. it'll always be a fixed length.

Reply:
changed to use substring

Reply:
fyi, changed back to using replaceFirst to handle case where the reverseChainParameterString might just be '_has'. If using substring, I'd have to first check the length of reverseChainParameterString before substringing past the expected initial '_has:' string.